include/optimization/BaseEigenproblemSolver.hpp

```diff
@@ -59,50 +55,30 @@ class BaseEigenproblemSolver {
 
 
     // GETTERS
-    /**
-     *  @return the diagonal of the matrix representation of the matrix whose eigenproblem is being solved
-     */
-    virtual Eigen::VectorXd get_diagonal() = 0;
     bool is_solved() const { return this->_is_solved; }
 
-
-    // GETTERS - EIGENPAIR
     std::vector<Eigenpair> get_eigenpairs() const;
```

Comment:
This line should return a const reference not a copy, this causes the sudden drop in accuracy.

Reply:
Sorry, my bad. Thanks for catching this!